[FSDP2][PT2D] enable torch.compile for compute in CI

weifengpy opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #122876
* #123021

34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype
* test_fully_shard_autograd::test_unused_forward_output

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k @rohan-varma

weifengpy committed:
[FSDP2][PT2D] enable torch.compile for compute in CI

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

awgu commented on test/distributed/_composable/fsdp/test_fully_shard_training.py:
```diff
@@ -52,14 +51,15 @@
 from torch.testing._internal.distributed.checkpoint_utils import with_temp_dir
 
 
-class TestFullyShardForwardInputs(FSDPTestMultiThread):
```
Are these changes from MTPG to normal NCCL PG meant for landing? I worry that this will (heavily) increase CI time. We want MTPG for these non-training-numerics related tests.

weifengpy replied:
for MTPG, I observed flakiness where dynamo compains about broken process pool. There is a recent post about the issue: https://github.com/pytorch/pytorch/issues/121197. we can discuss if this worth spending time to dig into

awgu replied:
I wonder if it is worth going through some of these tests and seeing if testing `torch.compile` would be valuable or not. Some of these seem like `torch.compile` for compute should not affect. As a bonus, I would hope we could keep the eager tests to use MTPG somehow 🤔

weifengpy replied:
MTPG works well after clearing dynamo cache with ``torch._dynamo.reset()`` before and after each test. this also avoid triggering dynamic shapes

awgu commented on torch/testing/_internal/common_fsdp.py:
```diff
@@ -1324,8 +1332,32 @@ def fully_shard_with_compiled_compute(*args, **kwargs):
+                    hook.__module__ == "torch.distributed._composable.fsdp._fsdp_state"
+                    and hook.__name__ == "_pre_forward"
+                ):
+                    args[0]._forward_pre_hooks[handle_id] = torch.compiler.disable(hook)
```
I wonder if we should land this `torch.compiler.disable()` for now. Without this, we may error during tracing the forward pre-hook in some cases (e.g. 2D with `DTensor`).

cc: @yf225

weifengpy replied:
will split out `torch.compile.disable` into a separate PR to discuss with @yf225

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype
* test_fully_shard_autograd::test_unused_forward_output

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype
* test_fully_shard_autograd::test_unused_forward_output

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype
* test_fully_shard_autograd::test_unused_forward_output

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

weifengpy committed:
Update on "[FSDP2][PT2D] enable torch.compile for compute in CI"


34 tests are eligible for torch.compile (call forward) 
* 70% passed
* 30% failed for 3 reasons

failure on 2D
* test_fully_shard_init.py::test_meta_device_2d_init
* test_fully_shard_training.py::test_train_parity_2d_mlp
* test_fully_shard_training.py::test_train_parity_2d_transformer_checkpoint_resume
* test_fully_shard_clip_grad_norm_.py::test_clip_grad_norm_2d

failure on composable AC
* test_fully_shard_training.py::test_train_parity_with_shared_params
* test_fully_shard_training.py::test_train_parity_with_activation_checkpointing
* test_fully_shard_comm.py::test_fully_shard_backward_prefetch
* test_fully_shard_frozen.py::test_train_mixed_requires_grad_per_group

failure on numerics
* test_fully_shard_mixed_precision.py::test_compute_dtype
* test_fully_shard_mixed_precision.py::test_reduce_dtype
* test_fully_shard_autograd::test_unused_forward_output

following 2 tests called forward but did not count as eligible since memory and time are different in compiler mode
* test_fully_shard_memory.py::test_fully_shard_training_memory
* test_fully_shard_overlap.py::test_fully_shard_training_overlap


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]
